go-controller/pkg/ovn/pods.go

```diff
@@ -232,7 +192,7 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) error {
 	}()
 
 	logicalSwitch := pod.Spec.NodeName
-	nodeSubnet, err := oc.waitForNodeLogicalSwitch(pod.Spec.NodeName)
+	err = oc.waitForNodeLogicalSwitch(logicalSwitch)
```

Comment:
This check `waitForNodeLogicalSwitch` seems redundant since we just return on line 184 above if the switch doesn't exist yet in the cache. I wasn't sure so have left it in there. I think it can be taken out.

Reply:
They aren't quite the same thing. We need to differentiate between:

1. we've seen this node already but chose not to assign a hostsubnet because of --no-hostsubnet-nodes=<label>
2. we've never seen this node before and need to wait for its hostsubnet to be allocated

The `val, ok := oc.logicalSwitchCache[pod.Spec.NodeName]; ok && val == nil` bit is checking if the node exists in the node map (eg we've seen it already) but chose not to assign a hostsubnet (val == nil).

The `err = oc.waitForNodeLogicalSwitch(logicalSwitch)` is for when the node doesn't exist yet in the node map (we haven't processed the node addition at all yet) because sometimes our Pod/Node events arrive out of order because Kubernetes and asynchronous event processing.

Reply:
OK, will leave both the checks in.